Copy notebook exports via nb header

RickGove opened the pull request:
Same as already approved PR, but much better functionality and looks

RickGove committed:
Added notebook location, and click to copy imports to notebook header

elis commented on packages/web/src/djitsu/components/notebook-header/index.js:
```diff
@@ -57,6 +58,9 @@ export const NotebookHeader = (props) => {
   const [state, actions] = useNotebook()
   const [documentMenuVisible, setDocumentMenuVisible] = useState()
 
+  let exports = []
```
what's the exports purpose?

RickGove replied:
Okay, so it has to be declared as an array, so that the exports can be iterated over. I will change it up though

elis commented on packages/web/src/djitsu/components/notebook-header/index.js:
```diff
@@ -291,6 +297,57 @@ export const NotebookHeader = (props) => {
   //   { hasMain, notebook, userProfile, notebookId, isOwner }
   // )
 
+  const fallbackCopyTextToClipboard = (text) => {
```
I'd extract this entire logic into a separate function or a custom hook - and i'd also check if there's a copy-to-clipboard package to use instead of hand-crafting it

elis commented on packages/web/src/djitsu/components/notebook-header/index.js:
```diff
@@ -291,6 +297,57 @@ export const NotebookHeader = (props) => {
+  }
+
+  const initCopy = () => {
+    let exportsAsString = ''
```
I'd prepare the string in advanced so it can be shown to the user - this can be a large blob of text on some occasions and not necessarily what the user expects.

RickGove committed:
Many improvements

- Added a modal where you can choose which exports
- Now only shows # of exports
- Copy is using an NPM package

elis commented on packages/web/src/djitsu/components/notebook-header/index.js:
```diff
@@ -73,6 +85,93 @@ export const NotebookHeader = (props) => {
 
+  const [showCopy, setShowCopy] = useState(false)
+
+  const CopyModal = () => {
```
Why is this defined here?

Not sure if needs to be defined here, but if you keep it make sure to wrap it in `useMemo` otherwise it will create a new component every time the header is rendered...

RickGove replied:
Moved it to main Render

RickGove committed:
Move modal render to main Render Fn

RickGove committed:
improved exports declaration, and exports text

RickGove committed:
add warning if NB not saved

RickGove committed:
Made all exports checked by default

RickGove committed:
Added enter and esc hotkeys
